### PR TITLE
chore(drive): improve quorum info update logs

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

Logs aren't looking the same on different nodes that complicates comparison 

## What was done?

- Sort deterministically the new quorum info log output
- Removed unnecessary big log data

## How Has This Been Tested?

None

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: packages/rs-drive-abci/src/execution/platform_events/core_based_updates/update_quorum_info/v0/mod.rs [hunk 1]
```diff
@@ -33,51 +33,60 @@ where
         core_block_height: u32,
         start_from_scratch: bool,
     ) -> Result<(), Error> {
-        if !start_from_scratch && core_block_height == block_platform_state.core_height() {
+        let _span = tracing::span!(Level::TRACE, "update_quorum_info", core_block_height).entered();
+
+        if start_from_scratch {
+            tracing::debug!("update quorum info from scratch up to {core_block_height}");
+        } else if core_block_height != block_platform_state.core_height() {
             tracing::debug!(
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-drive-abci/src/execution/platform_events/core_based_updates/update_quorum_info/v0/mod.rs:41]: Yeah, I have a field that we need in span. Do you suggest something specific to add?

File: packages/rs-drive-abci/src/execution/platform_events/core_based_updates/update_quorum_info/v0/mod.rs [hunk 2]
```diff
@@ -33,51 +33,60 @@ where
...
-        );
-
-        tracing::debug!(
-            method = "update_quorum_info_v0",
-            "new quorum_info {:?}",
-            quorum_info
-        );
+            .into_iter()
+            .map(|(quorum_type, quorum_list)| {
+                let sorted_quorum_list: BTreeMap<_, _> = quorum_list.into_iter().collect();
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-drive-abci/src/execution/platform_events/core_based_updates/update_quorum_info/v0/mod.rs:61]: BTeeMap compared with HashMap has additional overhead. Deterministic sorting is a Drive requirement but other apps that will use Core RPC client might not have it.

[the remark this answers is not in the document]
Comment (lklimek) [reply packages/rs-drive-abci/src/execution/platform_events/core_based_updates/update_quorum_info/v0/mod.rs:61]: core.rs which I mentioned is part of the  rs-drive-abci crate.

File: packages/rs-drive-abci/src/execution/platform_events/core_based_updates/update_quorum_info/v0/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (lklimek) [reply packages/rs-drive-abci/src/execution/platform_events/core_based_updates/update_quorum_info/v0/mod.rs:41]: old core height, new core height ? also quorum_hash might be useful to cross-check with tenderdash if it's easy to extract (but, I guess, on trace level)

[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-drive-abci/src/execution/platform_events/core_based_updates/update_quorum_info/v0/mod.rs:41]: Oh I see. I use params in the message but I didn't expose them as fields. Yes, my bad.

File: packages/rs-drive-abci/src/execution/platform_events/core_based_updates/update_quorum_info/v0/mod.rs [hunk 3]
```diff
@@ -33,51 +33,58 @@ where
...
+                        self.config.quorum_type
+                    ),
+                )))?;
 
         // Remove validator_sets entries that are no longer valid for the core block height
         block_platform_state
             .validator_sets_mut()
-            .retain(|key, _| quorum_info.contains_key(key));
+            .retain(|quorum_hash, _| {
+                let has_quorum = validator_quorums_list.contains_key::<QuorumHash>(quorum_hash);
```
Comment (QuantumExplorer) [inline packages/rs-drive-abci/src/execution/platform_events/core_based_updates/update_quorum_info/v0/mod.rs:71]: why add <QuorumHash>

Comment (QuantumExplorer) [reply packages/rs-drive-abci/src/execution/platform_events/core_based_updates/update_quorum_info/v0/mod.rs:71]: QuorumHash

File: packages/rs-drive-abci/src/execution/platform_events/core_based_updates/update_quorum_info/v0/mod.rs [hunk 4]
```diff
@@ -102,27 +110,39 @@ where
...
-            .extend(new_quorums.into_iter());
+            .extend(new_validator_sets);
 
+        // Sort all validator sets into deterministic order by core block height of creation
         block_platform_state
             .validator_sets_mut()
             .sort_by(|_, quorum_a, _, quorum_b| {
                 let primary_comparison = quorum_b.core_height().cmp(&quorum_a.core_height());
-                if primary_comparison == Ordering::Equal {
+                if primary_comparison == std::cmp::Ordering::Equal {
```
Comment (QuantumExplorer) [inline packages/rs-drive-abci/src/execution/platform_events/core_based_updates/update_quorum_info/v0/mod.rs:145]: why std::cmp::?

**Review by QuantumExplorer — CHANGES_REQUESTED** [review]

File: packages/rs-drive-abci/tests/strategy_tests/execution.rs [hunk 5]
```diff
@@ -686,7 +684,7 @@ pub(crate) fn continue_chain_for_strategy(
                     next_protocol_version,
                     change_block_height,
                 } = proposer_versions
-                    .get(&proposer.pro_tx_hash)
+                    .get::<ProTxHash>(&proposer.pro_tx_hash)
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-drive-abci/tests/strategy_tests/execution.rs:687]: IDE shows errors. It's annoying.

Screenshot 2023-10-06 at 13 48 42

File: packages/rs-drive-abci/src/execution/platform_events/core_based_updates/update_quorum_info/v0/mod.rs (hunk 4 above)
Comment (shumkov) [reply packages/rs-drive-abci/src/execution/platform_events/core_based_updates/update_quorum_info/v0/mod.rs:145]: There were usage `std::cmp::Ordering` and just `Ordering` in this module. I made it consistent with the full path: I found it more descriptive.

File: packages/rs-drive-abci/tests/strategy_tests/execution.rs (hunk 5 above)
[the remark this answers is not in the document]
Comment (QuantumExplorer) [reply packages/rs-drive-abci/tests/strategy_tests/execution.rs:687]: got it

**Review by QuantumExplorer — APPROVED** [review]

**Pull request merged**